test/spec/hierarchical-facets/no-refinement.js

```diff
@@ -59,29 +56,20 @@ test('hierarchical facets: no refinement', function(t) {
     }]
   }];
 
-  client.search = sinon
-    .stub()
-    .resolves(algoliaResponse);
+  client.search = jest.fn(function() {
+    return Promise.resolve(algoliaResponse);
+  });
 
   helper.setQuery('a').search();
   helper.once('result', function(content) {
-    var call = client.search.getCall(0);
-    var queries = call.args[0];
+    var queries = client.search.mock.calls[0][0];
     var hitsQuery = queries[0];
 
-    t.equal(queries.length, 1, 'we made one query');
-    t.ok(client.search.calledOnce, 'client.search was called once');
-    t.deepEqual(
-      hitsQuery.params.facets,
-      ['categories.lvl0'],
-      'first query (hits) has `categories.lvl0` as facets'
-    );
-    t.equal(
-      hitsQuery.params.facetFilters,
-      undefined,
-      'first query (hits) has no facet refinement refinement'
-    );
-    t.deepEqual(content.hierarchicalFacets, expectedHelperResponse);
-    t.end();
+    expect(queries.length).toBe(1);
+    expect(client.search).toHaveBeenCalledTimes(1);
+    expect(hitsQuery.params.facets).toEqual(['categories.lvl0']);
+    expect(hitsQuery.params.facetFilters).toBe(undefined);
```

Comment:
We can use [`toBeUndefined()`](https://jestjs.io/docs/en/expect#tobeundefined).

Reply:
Yes indeed. We can caught all those issues with the ESLint plugin for Jest. I didn't set it up because we use an (old) ESLint version that is not supported by the plugin. The first step is to have the tests pass, then we can improve them.